qutip/core/data/eigen.py

```diff
@@ -225,6 +227,15 @@ def eigs_csr(data, isherm=None, vecs=True, sort='low', eigvals=0,
     isherm = isherm if isherm is not None else _isherm(data)
     evals, evecs = _eigs_csr(data.as_scipy(), isherm, vecs, eigvals,
                              num_large, num_small, tol, maxiter)
+
+    if vecs and isherm:
+        i = 0
+        while i < len(evals):
+            num_degen = np.sum(np.abs(evals - evals[i]) < (2 * tol or 1e-14))
+            for k, l in combinations(range(num_degen), 2):
+                _orthogonalize(evecs[:, i+l], evecs[:, i+k])
```

Comment:
```suggestion
            # orthogonalize vectors 1 .. k with respect to the first, then 2 .. k with respect to the second, and so on.
            # relies on both the order of each pair and the ordering of pairs returned by combinations. 
            for k, l in combinations(range(num_degen), 2):
                _orthogonalize(evecs[:, i+l], evecs[:, i+k])
```
The ordering requirements used here are nice, but also non-obvious, so I suggested a comment.